src/main/java/org/opensearch/knn/quantization/models/quantizationState/QuantizationStateCacheManager.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.knn.quantization.models.quantizationState;
+
+import lombok.AccessLevel;
+import lombok.NoArgsConstructor;
+import org.opensearch.knn.index.codec.KNN990Codec.KNNQuantizationStateReader;
+
+import java.io.IOException;
+
+@NoArgsConstructor(access = AccessLevel.PRIVATE)
+public final class QuantizationStateCacheManager {
+
+    private static volatile QuantizationStateCacheManager instance;
+
+    /**
+     * Gets the singleton instance of the cache.
+     * @return QuantizationStateCache
+     */
+    public static QuantizationStateCacheManager getInstance() {
+        if (instance == null) {
+            synchronized (QuantizationStateCacheManager.class) {
+                if (instance == null) {
+                    instance = new QuantizationStateCacheManager();
+                }
+            }
+        }
+        return instance;
+    }
+
+    public synchronized void rebuildCache() {
+        QuantizationStateCache.getInstance().rebuildCache();
+    }
+
+    /**
+     * Retrieves the quantization state associated with a given field name.  Reads from cache first, then from disk if necessary.
+     * @param quantizationStateReadConfig information required from reading from off-heap if necessary
+     * @return The associated QuantizationState
+     */
+    public QuantizationState getQuantizationState(QuantizationStateReadConfig quantizationStateReadConfig) throws IOException {
+        String fieldName = quantizationStateReadConfig.getFieldInfo().getName();
+        QuantizationState quantizationState = QuantizationStateCache.getInstance().getQuantizationState(fieldName);
+        if (quantizationState == null) {
+            quantizationState = KNNQuantizationStateReader.read(quantizationStateReadConfig);
+            addQuantizationState(fieldName, quantizationState);
+        }
+
+        return quantizationState;
+    }
+
+    /**
+     * Adds or updates a quantization state in the cache.
+     * @param fieldName The name of the field.
+     * @param quantizationState The quantization state to store.
+     */
+    private void addQuantizationState(String fieldName, QuantizationState quantizationState) {
```

Comment:
I think this function has to be public so that , from NativeReader we can load all cache.